src/vt_flash/newton.jl

```diff
@@ -55,19 +55,39 @@ function backtracking_line_search(
 end
 
 """
-    newton(f, grad!, hess!, x₀[; gtol, maxiter, constrain_step])
+    newton(f, grad!, hess!, x₀[; gtol=1e-6, maxiter=200, constrain_step, convcond])
 
-Find constrained minimum of `f(x)::Real` using Newton's solver with line search
+Find (un)constrained minimum of `f(x)::Real` using Newton's solver with line search
 and hessian modified by Cholesky factorization.
-`grad!(g, x)` must update and return `g`radient of `f` at `x`.
-`hess!(H, x)` must update and return `H`essian matrix of `f` at `x`.
 
 The optimization starts at `x₀` performing no more than `maxiter` Newton's steps.
-The steps finish when norm of gradient is ≤ `gtol`.
-New point `xnew <- x + αmax * d` is constrained in Newton's `d`irection by
-`constrain_step(x, d) -> α::Real`, `αmax = min(1.0, α)`.
+
+New point `xnew ← x + αmax * d` is constrained in Newton's `d`irection by
+`constrain_step(x, d) -> α::Real`.
+By default, the constrains are disabled.
+
+The *end of optimization* is controlled by `gtol` and `convcond`.
+When `convcond` is omitted, the optimization ends when 2-norm of gradient is ≤ `gtol`.
+When `convcond(x, g) -> Bool` is provided, it's called before checking of gradient's norm.
```

Comment:
Makes sense to make a more generic
`convcond(x, xpre, f, fpre, g) -> Bool` and omit gradient check entirely if it is provided.